### PR TITLE
docs: add bitbucket app password requirements

### LINKED ISSUE #1119: When working with BitBucket credentials are never reused
### Version

2.0.886+ea93cb5158

### Operating system

Windows

### OS version or distribution

Windows 10

### Git hosting provider(s)

Bitbucket Cloud

### Other hosting provider

_No response_

### (Azure DevOps only) What format is your remote URL?

None

### Can you access the remote repository directly in the browser?

Yes, I can access the repository

### Expected behavior

Not having to resupply credentials on each pull/push.

### Actual behavior

On one of our computers we had problems with pulling from BitBucket. That has been fixed, but now the App-password must be entered each time a pull or push is made.
We have tried uninstalling Git for Windows (which includes git-credential-manager). Also deleting "[...]\Program Files\git" and deleting .git files in the user directory and removing BitBucket credentials under windows-credentials. In other words allowing for a completely fresh install.

We then install the latest version of Git for Windows (2.39.2).
We try to do a pull, and there are two options. By browser or by password. By browser sends the user to a page where you have to grant access, but that just leads one to an URL at localhost which doesn't work. Hence we use the password option instead. Username and App-password is entered and the pull works. However, during the next pull the credentials have to be supplied again. Under windows-credentials there are now credentials for BitBucket, but apparently they are not used.

I did see the closed issue https://github.com/GitCredentialManager/git-credential-manager/issues/1116 but running "git config --global credential.helper wincred" did not help.

### Logs

_No response_

### PR DESCRIPTION
**Pull request by ldennington**
Separate instructions for authentication into a new `bitbucket-authentication` file. For Password/Token authentication, add a callout that, when using an App Password, account read permissions are required.

Closes #1119

### CODE REVIEW

File: docs/img/app-password.png
Comment (mjcheetham) [inline docs/img/app-password.png:1]: Do we need Workspace membership (Read)?

File: docs/bitbucket-development.md
Comment (mjcheetham) [inline docs/bitbucket-development.md:1]: This file is more geared towards development of the Bitbucket host provider. Do we want to pull it out in to two separate docs? One dev focused and one user focused?

**Review by mjcheetham — APPROVED** [review]

File: docs/img/app-password.png
Comment (ldennington) [reply docs/img/app-password.png:1]: Yes. I tried generating a token without these minimal permissions and authentication failed. I can call that out in the text.

File: docs/bitbucket-development.md
Comment (ldennington) [reply docs/bitbucket-development.md:1]: I'm fine with creating a separate doc. I will just plan to pull the whole "Authentication User Experience" section into it.

**Review by mjcheetham — APPROVED** [review]

**Pull request merged**